proposals/VK_EXT_present_timing.adoc

```diff
@@ -0,0 +1,470 @@
+
+When supported, `presentAtAbsoluteTime` allows applications to specify an absolute time (in the selected time domain) with each `vkQueuePresentKHR` call.
+Similarly, `presentAtRelativeTime` allows applications to specify a relative time with each `vkQueuePresentKHR`. See <<scheduling>>.
+
+These features are also advertised for each `VkSurfaceKHR` object with:
+
+[source,c]
+----
+typedef struct VkSurfacePresentTimingCapabilitiesEXT {
+    VkStructureType           sType;
+    void*                     pNext;
+    VkBool32                  presentTimingSupported;
+    VkBool32                  presentAtAbsoluteTimeSupported;
+    VkBool32                  presentAtRelativeTimeSupported;
+    VkPresentStageFlagsEXT    presentStageQueries;
+    VkPresentStageFlagsEXT    presentStageTargets;
+} VkSurfacePresentTimingCapabilitiesEXT;
+----
+
+In addition of the present timing and present scheduling features, surfaces also advertise which <<present_stages>> are available to query timing and schedule presents for.
+
+=== Present stages [[present_stages]]
+
+It is difficult to define "presentation" while satisfying all implementations, platforms or even display technologies. Thus, this proposal defines present stages: a set of well-defined discrete steps within typical present pipelines.
+
+[source,c]
+----
+typedef enum VkPresentStageFlagBitsEXT {
+    VK_PRESENT_STAGE_IMAGE_HANDOFF_BIT_EXT = 0x00000001,
+    VK_PRESENT_STAGE_IMAGE_LATCHED_BIT_EXT = 0x00000002,
+    VK_PRESENT_STAGE_IMAGE_FIRST_PIXEL_OUT_BIT_EXT = 0x00000004,
+    VK_PRESENT_STAGE_IMAGE_FIRST_PIXEL_VISIBLE_BIT_EXT = 0x00000008,
+} VkPresentStageFlagBitsEXT;
+----
+
+When enabling timing feedback for a swapchain, a set of present stages is specified to inform the implementation that timing for all those stages is desired. See <<statistics>>.
+
+Similarly, when using `presentAtAbsoluteTime` feature to schedules presents at specific times, a present stage must be specified as a target.
+// XXX: What about relative timing?
+
+* `VK_PRESENT_STAGE_IMAGE_HANDOFF_BIT_EXT` corresponds to the step after which the presentation request has been passed to a system-specific entity of the presentation engine that is outside the control of the Vulkan implementation. Because the implementation itself is responsible for this step, it is able to record this time with high accuracy and low latency.
+* `VK_PRESENT_STAGE_IMAGE_LATCHED_BIT_EXT` is the step after which the image associated with the presentation request has been latched by the presentation engine to create the presentation of a future refresh cycle. For example, in a flip-model scenario, this is the time the presentation request's image has been selected for the next refresh cycle.
```

Comment:
I'm not sure "low latency" applies to Android.  Maybe better to state that this is for some platforms.